grpc-gcp/src/main/java/com/google/cloud/grpc/GcpMultiEndpointChannel.java

```diff
@@ -318,7 +518,15 @@ public synchronized void setMultiEndpoints(List<GcpMultiEndpointOptions> meOptio
     defaultMultiEndpoint = multiEndpoints.get(meOptions.get(0).getName());
 
     // Remove obsolete multiendpoints.
-    multiEndpoints.keySet().removeIf(name -> !currentMultiEndpoints.contains(name));
+    Iterator<String> iter = multiEndpoints.keySet().iterator();
+    while (iter.hasNext()) {
```

Comment:
I am not familiar with Java, so this might be a stupid question. But since this is a loop, do you need to update iter? Like iter = iter.Next();

Reply:
no, there is no need in that. When we call `iter.next()` it advances the current item pointer and retrieves the item.